CMake/build.cmake

```diff
@@ -13,23 +15,37 @@ if(NOT CMAKE_BUILD_TYPE)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+# SENSEI should always be position independent
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+# SENSEI requires minimum C++11
+if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 11)
+  set(CMAKE_CXX_STANDARD 11)
+endif ()
+
 if (NOT MSVC)
-  if (NOT CMAKE_CXX_FLAGS)
-  set(tmp "-fPIC -std=c++11 -Wall -Wextra")
+  string(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra")
+
+  # Need to explicitly set optimization level to -O3 because some systems default to -O2
+  set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
+  set(optimization_flags " -mtune=native; -march=native")
+  foreach (flag ${optimization_flags})
+    check_cxx_compiler_flag(${flag} has_flag)
+    if(has_flag)
+      string(APPEND CMAKE_CXX_FLAGS_RELEASE ${flag})
+    endif ()
+  endforeach ()
+
   if (BUILD_STATIC_EXECS)
-    set(tmp "${tmp} -static -static-libgcc -static-libstdc++ -pthread -Wl,-Bstatic")
-  endif()
-  if ((APPLE) AND ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang"))
-    set(tmp "${tmp} -stdlib=libc++")
-  endif()
-  if ("${CMAKE_BUILD_TYPE}" MATCHES "Release")
-    set(tmp "${tmp} -O3 -march=native -mtune=native")
-  endif()
-  set(CMAKE_CXX_FLAGS "${tmp}"
-    CACHE STRING "SENSEI build defaults"
-    FORCE)
-  endif()
-endif()
+    string(APPEND CMAKE_CXX_FLAGS "-static -static-libgcc -static-libstdc++ -pthread -Wl,-Bstatic")
+  endif ()
+endif ()
+
+if (APPLE)
+  if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
```

Comment:
Clang and/or AppleClang?

Reply:
I find the stdlib control in this file generally a bit curious: this selection should be up to the user/package manager usually, otherwise you might not integrate well with a larger science app software stack that might select another stdlib than you enforce here.

Reply:
this was necessary in the past to compile on Mac using Clang and C++11. Admittedly it was a long time ago, and the code was propagated forward. Apparently it is no longer necessary and if it could be removed at this point, OK to do so. This would also mean not having to FORCE the cache variable to ensure that we always compile. Another improvement.

Reply:
on second thought, would like to verify that brew'd clang  wasn't the source of the need for adding that flag.